modules/internal/ChapelArray.chpl

```diff
@@ -2027,7 +2030,28 @@ module ChapelArray {
     proc tail(): this._value.eltType {
       return this[this.domain.high];
     }
-
+    pragma "no doc"
+    proc resizeAllocRange(r: range, d2: domain, factor=arrayAsVecGrowthFactor, direction=1, grow=1) {
+      const lo = r.low,
+            hi = r.high,
+            size = hi - lo + 1;
+      var newSize: int;
+      if grow > 0 {
+        newSize = max(size+1, (size*factor):int); // Always grow by at least 1.
+        if direction > 0 {
+          return lo..#newSize;
+        } else {
+          return ..hi#-newSize;
+        }
+      } else {
+        // shrink to match the d2 bound on the side indicated by direction
+        if direction > 0 {
+          return lo..d2.high;
+        } else {
+          return d2.low..hi;
+        }
+      }
+    }
     /* Add element ``val`` to the back of the array, extending the array's
        domain by one. If the domain was ``{1..5}`` it will become ``{1..6}``.
      */
```

Comment:
While there, I suggest extending this comment (and similar comments on later functions) with:

The array must be a rectangular 1-D array; its domain must be non-stridable and not shared with other arrays."